pkg/monitor/nodedetails/audit_log_summarizer.go

```diff
@@ -0,0 +1,417 @@
+package nodedetails
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+)
+
+// every audit log summarizer is not threadsafe. The idea is that you use one per thread and
+// later combine the summarizers together into an overall summary
+type AuditLogSummary struct {
+	lineReadFailureCount    int
+	requestCounts           RequestCounts
+	perUserRequestCount     map[string]*PerUserRequestCount
+	perResourceRequestCount map[schema.GroupVersionResource]*PerResourceRequestCount
+	perStatusRequestCount   map[int32]*PerStatusRequestCount
+}
+
+type RequestCounts struct {
+	requestStartedCount       int
+	requestFinishedCount      int
+	clientFailedRequestCount  int
+	serverFailedRequestCount  int
+	perHTTPStatusRequestCount map[int32]int
+}
+
+type PerStatusRequestCount struct {
```

Comment:
Nit: `PerHTTPStatusRequestCount` to make it explicit clear what status it is about, which was my first question when I read this struct name.